js/api/index.js

```diff
@@ -0,0 +1,87 @@
+/**
+ * A new version of the functions to interact with the Proxy API.
+ *
+ * @flow
+ */
+
+import { apiUrlPrefix } from '../config'
+
+export type ApiFetchSuccess<T> = {
+  isError: false,
+  result: T
+}
+
+export type ApiFetchFailure = {
+  isError: true,
+  error: Error
+}
+
+export type ApiFetchResult<T> = ApiFetchSuccess<T> | ApiFetchFailure
+
+// Utility type to add a required version property
+export type Versionable = {
+  version: number
+}
+
+/**
+ * A type that make alla fields of T optional than add `version` as the only
```

Comment:
A type that makes all fields of type `T` optional, then adds `version` as the only
required field. This type is used mostly to update an API entity.